transformer_engine/debug/features/api.py

```diff
@@ -231,6 +233,9 @@ def inspect_tensor(
         tp_group: torch.distributed.ProcessGroup,
     ) -> None:
         """
+        This is legacy call, we advise to use *inspect_tensor_all* and *inspect_tensor_all_enabled* instead.
```

Comment:
If we don't want users to call `inspect_tensor` and other legacy functions, we should raise a deprecation warning.

Also, it seems like a shame to so quickly abandon these function names and replace them with clunky `_all` variants. Is there a way to maintain backward compatibility? Maybe have kwargs like `high_precision_tensors` and `quantized_tensors`, and have `quantized_tensors=False` by default.